rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java

```diff
@@ -679,20 +702,23 @@ protected Literal parseQuotedLiteral() throws IOException, RDFParseException, RD
 
 			// Read datatype
 			Value datatype = parseValue();
-			if (datatype instanceof IRI) {
-				return createLiteral(label, null, (IRI) datatype, getLineNumber(), -1);
-			} else {
+			if (datatype instanceof Literal || datatype instanceof BNode) {
```

Comment:
@ansell new simplified logic. I distinguish between the datatype being an invalid IRI vs not being an IRI at all (e.g. a literal or bnode value instead). The latter should always result in a fatal error. The former only based on whether VERIFY_URI_SYNTAX is set. See the unit tests to understand how this behaves in practice.

Reply:
That logic looks okay to me and more maintainable. The fixed set of negative `instanceof` checks aren't an issue because the RDF model isn't going to be changing anytime soon (and `instanceof` checks are relied on throughout the whole codebase anyway)